l3build-check.lua

```diff
@@ -696,22 +696,22 @@ function runtest(name, engine, hide, ext, pdfmode, breakout)
   cp(lvtfile, fileexists(testfiledir .. "/" .. lvtfile)
     and testfiledir or unpackdir, testdir)
   local checkopts = checkopts
-  local engine = engine or stdengine
+  engine = engine or stdengine
   local binary = engine
-  local format = gsub(engine,"tex$",checkformat)
+  local format_a = gsub(engine,"tex$",checkformat)
   -- Special binary/format combos
   if specialformats[checkformat] and next(specialformats[checkformat]) then
     local t = specialformats[checkformat]
-    if t[engine] and next(t[engine]) then
-      local t = t[engine]
```

Comment:
local t hides outer t

Reply:
`t` is a throw-away variable anyway, so I'm not sure I see the issue with hiding the 'outer' `t`. If we want another name, probably `T` or `tbl` or similar.

Reply:
With the `t_ngn` variable, the problem has disappeared